frontend/src/slicer/productSlice.ts

```diff
@@ -1,53 +1,119 @@
-import {createAsyncThunk, createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {createAsyncThunk, createSlice, Dispatch, PayloadAction} from '@reduxjs/toolkit';
 import {RootState} from '../app/store';
-import { IProductsState } from '../interfaces/IStates';
-import {getAllProducts} from '../services/productService';
+import { IProductsState} from '../interfaces/IStates';
 import {getErrorMessage} from "./errorSlice";
 import {IResponseGetAllProducts} from "../interfaces/IApiResponse";
+import {
+    getAll as apiGetAll,
+    getOne as apiGetOne,
+    create as apiCreate,
+    edit as apiEdit,
+    del as apiDelete
+} from '../services/apiService'
+import {IProduct} from "../interfaces/IProduct";
 
 
-const initialState:IProductsState = {
-    products:[],
+const initialState: IProductsState = {
+    products: [],
     currentProduct: undefined,
     pending: false,
 }
+export const handleError = (status:number, statusText: string, dispatch: Dispatch) => {
+    if (status !== 200) {
+        dispatch(getErrorMessage({status, statusText}))
+    }
+}
 
 
-export const getProducts = createAsyncThunk(
-    'getProducts',
+export const getAllProducts = createAsyncThunk<IResponseGetAllProducts, void, { state: RootState, dispatch: Dispatch }>(
+    'getAll',
     async (_, {getState, dispatch}) => {
-        //@ts-ignore
+        console.log(Date.now());
         const token = getState().authentication.token;
-        const {data, status, statusText} = await getAllProducts(token);
-        if (status !== 200) {
-            dispatch(getErrorMessage({status, statusText}))
-        }
+        const {data, status, statusText} = await apiGetAll("product", token);
+        handleError(status, statusText, dispatch);
         return {data, status, statusText}
     }
 )
 
+export const getOneProduct = createAsyncThunk<IResponseGetAllProducts, string, { state: RootState, dispatch: Dispatch }>(
+    'getOne',
+    async (id, {getState, dispatch}) => {
+        const token = getState().authentication.token
+        const {data, status, statusText} = await apiGetOne("product", token, id);
+        handleError(status, statusText, dispatch);
+        return {data, status, statusText}
+    }
+)
 
+export const createProduct = createAsyncThunk<IResponseGetAllProducts, void, { state: RootState, dispatch: Dispatch }>(
+    'create',
+    async (_, {getState, dispatch}) => {
```

Comment:
What's this underscore?

Reply:
underscore is a parameter that doesn't get used. The way this method is constructed it will always give me a parameter. But in this function I don't need one, therefor the void type declaration two lines above. Since I need the ThunkApi Object (the getState and dispatch) just next to it, I have to declare the first parameter as well. 
To name it underscore is a convention to show that it doesn't get used